### ISSUE TITLE
Chatapp with AMP

### ISSUE THREAD

**Issue by vishalvanpariya** [ampproject/amphtml#27640]
Hello,

I am creating a chatbot using AMP.
so in a chatbot, I am fetching data from API.
I showing those data in the list but it should come one by one with user interaction.
and I sending users data to API.

adding one image here:
![Capture](https://user-images.githubusercontent.com/27823328/78781509-04d47100-79be-11ea-888e-30dd4ce17271.PNG)

the chatbot will be like above so please help me with this to create it.

Thanks

Vishal

**Comment by wassgha** [comment]
@caroqliu do you know if this is possible with `amp-list`?

**Comment by caroqliu** [comment]
Two options with amp-list: You should be able to resize the list with [dynamic content using `[is-layout-container]` or `changeToLayoutContainer`](https://amp.dev/documentation/components/amp-list/#[is-layout-container]-(optional)) on user interaction. Alternatively, you can define a fixed height amp-list and scroll content when its height exceeds its container. This is a UX I've seen for chat apps before.

As another option, @alanorozco gave a talk about how to build an amp-script chat box from the 2019 AMP Contributor Summit. Here is the [talk](https://www.youtube.com/watch?v=kLgnRleeP3s) and related [source code](http://bit.ly/amp-script-chat).

**Comment by amanintech** [comment]
Example with WebSocket and AMP.setState()
```
<amp-script width="1" height="1" script="webSocketDemo"> </amp-script>

<!--
  <amp-state> doesn't support WebSocket URLs in its "src" attribute,
  but we can use <amp-script> to work around it. :)
-->
<script type="text/plain" target="amp-script" id="webSocketDemo">
  const socket = new WebSocket('wss://websocket.example');
  socket.onmessage = event => {
    AMP.setState({socketData: event.data});
  };
</script> 
```


There is an example where you can use <amp-script> and then use WebSockets to fetch the latest messages and update them.

Here is the [link ](https://amp.dev/documentation/components/amp-script/#example-with-websocket-and-amp.setstate())to the documentation

**Comment by caroqliu** [comment]
Closing this issue because there have been a few resolutions suggested.
